tests/AbstractRepositoryTest.php

```diff
@@ -718,4 +774,100 @@ public function testFindExpectsStringPath()
 
         $repo->find(new \stdClass());
     }
+
+    public function testLinkDirectory()
+    {
+        $repo = $this->createAutoresolveLinksRepository();
+
+        // Real directory
+        /** @var DirectoryResource $realdir */
+        $realdir = $repo->get('/webmozart/realdir');
+
+        $this->assertInstanceOf('Puli\Repository\Resource\DirectoryResource', $realdir);
+        $this->assertPathsAreEqual(__DIR__.'/Fixtures/dir1', $realdir->getFilesystemPath());
```

Comment:
These test methods should be split up in separate test methods. A test method generally consists of several steps:

* fixture setup
* execution of the test
* assertions

The fact that you repeat these steps here shows that this test is testing too much (test smell). The result in practice is that the test doesn't document well what functionality it is testing.